Remove references to deprecated Randomize XBlock

catong opened the pull request:
## [DOC-2695](https://openedx.atlassian.net/browse/DOC-2695)

Remove references to deprecated Randomize XBlock and methods of creating randomized problems by exporting a course and editing XML files. Provide pointers instead to supported Randomized Content Blocks. (https://openedx.atlassian.net/browse/TNL-3203, https://openedx.atlassian.net/browse/TNL-4109)

### Date Needed: Wednesday Feb 24

The feature is deprecated as of Wed Feb 24 release.

### Reviewers

Possible roles follow. PR submitter checks the boxes after each reviewer
finishes and gives :+1:. 

- [x] Subject matter expert: @symbolist 
- [x] Doc team review (sanity check): @lamagnifica @pdesjardins or @srpearce 
- [ ] Product review: @scottrish 

FYI 
@jaakana
@sstack22

### Draft HTML Version

- [x] http://draft-br-deprecated-randomize.readthedocs.org/en/latest/course_components/create_problem.html#problem-randomization

### Testing

- [x] Ran ./run_tests.sh without warnings or errors



### Post-review

- [x] Add description to release notes task as a comment
- [x] Squash commits

edx-webhook commented:
## [DOC-XXXX](https://openedx.atlassian.net/browse/DOC-XXXX)

Add a description of your changes.


### Date Needed (optional)

If the release date of a feature is known or estimated, provide it to give
reviewers guidance on turnaround time.


### Reviewers

Possible roles follow. PR submitter checks the boxes after each reviewer
finishes and gives :+1:. 

- [ ] Subject matter expert: 
- [ ] Subject matter expert: 
- [ ] Doc team review (sanity check/copy edit/dev edit): 
- [ ] Product review:
- [ ] Partner support: 
- [ ] PM review: 

FYI: Tag anyone else who might be interested in this PR here.


### Testing

- [ ] Ran ./run_tests.sh without warnings or errors


### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

 
### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here


### Post-review

- [ ] Add description to release notes task as a comment
- [ ] Squash commits

lamagnifica commented on en_us/shared/course_components/create_problem.rst:
```diff
@@ -683,22 +683,18 @@ Problem Randomization
+You can provide different learners with different problems by using randomized
+content blocks, which randomly draw problems from pools of problems stored in
+content libraries. For more information, see
+:ref:`partnercoursestaff:Randomized Content Blocks`.
```
are content blocks only available for partners? should this entire paragraph have an "only" directive?
otherwise, no need for the partnercoursestaff indicator, right?

lamagnifica replied:
also, does this mean changes are also needed to the OLX guide?

catong replied:
Content blocks are available to both partners and open edX. Thanks for the reminder re OLX!

lamagnifica commented on en_us/shared/course_components/create_problem.rst:
```diff
@@ -736,86 +732,14 @@ contains. To find a unit, look in the **Vertical** directory.
+You can provide different learners with different problems by using randomized
+content blocks, which randomly draw problems from pools of problems stored in
+content libraries. For more information, see
+:ref:`partnercoursestaff:Randomized Content Blocks`.
```
ditto on the specific guide indication.
is there a reason to include this section? couldn't we just release note that this feature is no longer supported? or is it now deprecated, and will be removed at some point in the future, so this has to linger around here... forever...

catong replied:
I thought it was worth keeping this section for now for course teams who are searching for info on how to create randomized problems. I can re-evaluate/re-work all the content relating to randomizing problems when the code is actually removed from edx-platform.

lamagnifica commented:
See OLX info here: http://edx.readthedocs.org/projects/edx-open-learning-xml/en/latest/problem-xml/create_problem.html?highlight=randomize#problem-randomization

symbolist commented:
Looks good to me. :+1:

catong commented:
@lamagnifica Please take a further look. I updated the OLX guide to mirror the content in the shared file. In both sources, in the Randomization setting topic, I moved the note explaining the difference between variable randomization and problem randomization up to immediately after the title. Also moved the comment out from between 2 bullets where it was affecting spacing in output.

lamagnifica commented on en_us/olx/source/problem-xml/create_problem.rst:
```diff
@@ -511,61 +537,47 @@ answers for all the problems in the component appear.
 
-#. Open the ``vertical`` element that contains the randomized problems.
+.. note:: Creating randomized problems by exporting your course and editing
+   some of your course's XML files is no longer supported.
```
it might be worth stating this another time up at line 544ish, before they get to actually creating...

lamagnifica commented on en_us/olx/source/problem-xml/create_problem.rst:
```diff
@@ -511,61 +537,47 @@ answers for all the problems in the component appear.
 
-   .. code-block:: xml
+.. include:: ../../../shared/exercises_tools/Section_adding_tooltip.rst
 
```
why would this section be relevant in place of the problem example originally shown?

catong replied:
It's not in place of the example. It was originally at line 569 and just got moved up due to removal of the example.

lamagnifica replied:
got it.

lamagnifica commented on en_us/olx/source/problem-xml/create_problem.rst:
```diff
@@ -511,61 +537,47 @@ answers for all the problems in the component appear.
-   that you want to randomize.
 
-   .. code-block:: xml
+.. _Using edX Insights: http://edx.readthedocs.org/projects/edx-insights/en/latest/
```
this should be in the links file

lamagnifica commented on en_us/shared/course_components/create_problem.rst:
```diff
@@ -418,6 +418,13 @@ answers, the learner's score is 0.5 out of 2 points.
 Randomization
 ===============
 
+.. note:: The **Randomization** setting serves a different purpose from
```
possibly, "This **Randomization** setting" or "The **Randomization** setting described by this topic..." for clarity?
(same would apply to line 266 of the other file)

catong replied:
I'm inclined to leave this as is. Saying "The Randomization setting described by this topic" implies that there's another Randomization setting described elsewhere -- adding to the potential confusion.

lamagnifica commented:
I'm confused by that tooltip section, and had a few suggestions, but otherwise this looks good to go! :+1:

catong commented:
@lamagnifica The tooltip section is an included section that always followed the Randomization section. It just got moved up due to removal of the obsolete content.

catong commented:
Merging https://github.com/edx/edx-documentation/pull/884
